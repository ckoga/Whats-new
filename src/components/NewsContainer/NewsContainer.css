.newsContainer {
  grid-area: section;
  columns: 2;
}