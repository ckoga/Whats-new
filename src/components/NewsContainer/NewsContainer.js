import React from 'react';
import './NewsContainer.css'
import NewsArticle from '../NewsArticle/NewsArticle';

// NEWSCONTAINER COMPONENT CODE GOES HERE
const NewsContainer = ({articles}) => {
  return (
    articles.map((article, i) => <NewsArticle key={`article ${i}`} news={article}/>)
  );
}


export default NewsContainer;