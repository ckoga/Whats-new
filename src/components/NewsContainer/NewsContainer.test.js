import React from 'react';
import { shallow } from 'enzyme';
import NewsContainer from './NewsContainer';

describe('NewsContainer', () => {
  it('should be able to render NewsContainer to the DOM', () => {
    const wrapper = shallow(<NewsContainer 
      articles={[
      {id: 1, headline: 'Small News', img: 'pic1', description: 'A little bit of news', url: 'www.example.com'},
      {id: 2, headline: 'Medium News', img: 'pic2', description: 'Some news', url: 'www.example2.com'},
      {id: 3, headline: 'Big news', img: 'pic3', description: 'A lot of news', url: 'www.example3.com'}
      ]}
    />)

    expect(wrapper).toMatchSnapshot();
  })
})