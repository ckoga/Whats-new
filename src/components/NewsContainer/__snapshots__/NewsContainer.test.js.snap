// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsContainer should be able to render NewsContainer to the DOM 1`] = `
Array [
  <NewsArticle
    key="article 0"
    news={
      Object {
        "description": "A little bit of news",
        "headline": "Small News",
        "id": 1,
        "img": "pic1",
        "url": "www.example.com",
      }
    }
  />,
  <NewsArticle
    key="article 1"
    news={
      Object {
        "description": "Some news",
        "headline": "Medium News",
        "id": 2,
        "img": "pic2",
        "url": "www.example2.com",
      }
    }
  />,
  <NewsArticle
    key="article 2"
    news={
      Object {
        "description": "A lot of news",
        "headline": "Big news",
        "id": 3,
        "img": "pic3",
        "url": "www.example3.com",
      }
    }
  />,
]
`;
